scripts/mkcloud

```diff
@@ -396,13 +384,17 @@ function onhost_deploy_image()
     safely qemu-img convert -p $image $disk
     popd
 
-    if fdisk -l $disk | grep -q $offset ; then
+    # only resize if we have a 2nd partition with a rootfs
+    if fdisk -l $disk | grep -q "admin2 *\* *.*83 *Linux" ; then
         # make a bigger partition 2
         echo -e "d\n2\nn\np\n2\n\n\na\n2\nw" | fdisk $disk
-        local loopdevice=$(losetup --find --show -o $(( $offset * 512 )) $disk)
-        safely fsck -y -f $loopdevice
-        safely resize2fs $loopdevice
-        losetup -d $loopdevice
+        part2=$(kpartx -asv $disk|perl -ne 'm/add map (\S+2) / && print $1')
+        test -n "$part2" || complain 31 "failed to find partition #2"
+        local bdev=/dev/mapper/$part2
+        safely fsck -y -f $bdev
+        safely resize2fs $bdev
+        sleep 1 # time for dev to become unused
```

Comment:
that does not help. it still errors with
device-mapper: remove ioctl on mkcvg0-g15.admin2 failed: Device or resource busy